AIDriver.lua

```diff
@@ -907,6 +907,41 @@ function AIDriver:updateLights()
 	end
 end
 
+function AIDriver:updateAILights(superFunc)
+	if self.cp and self.cp.driver and self:getIsCourseplayDriving() and self.spec_lights then
+		if self.cp.driver:shouldLightsBeUsedForEnvironment() then --fall back to base class AIDriver for this
+			self.cp.driver:setLightsMask(self)
+		elseif superFunc ~= nil then
+			superFunc(self)
+		end
+	elseif superFunc ~= nil then
+		superFunc(self)
+	end
+end
+Lights.updateAILights = Utils.overwrittenFunction(Lights.updateAILights , AIDriver.updateAILights)
+
+function AIDriver:shouldLightsBeUsedForEnvironment()
+	-- How Giants decides lights in Lights:updateAILights
+	local dayMinutes = g_currentMission.environment.dayTime / (1000 * 60)
+	local nightTime = (dayMinutes > g_currentMission.environment.nightStartMinutes or dayMinutes < g_currentMission.environment.nightEndMinutes)
+	local rainScale = g_currentMission.environment.weather:getRainFallScale()
+	local timeSinceRain = g_currentMission.environment.weather:getTimeSinceLastRain()
+	local raining = rainScale > 0
+
+	local shouldLightsBeUsed = (nightTime or raining)
+
+	return shouldLightsBeUsed
+end
+
+function AIDriver:setLightsMask(vehicle)
+	local x,y,z = getWorldTranslation(vehicle.rootNode);
```

Comment:
Using the position for example in Mode 2 will result in switching between the lights whenever the vehicle leaves the field during maneuvering. Is this what we want?

Reply:
That's what I was going for. Idea being Mode 2 is a little bit open ended, and when drivers are on fields we have no reason to be using just headlights, and when on roads, we wouldn't want to blind any other drivers. Obviously, this is all just for role playing. Any solution here will work. The open ended nature of Mode 2 makes me wonder if there should be a setting a user can set for what headlight mode they want to be using. But overall, I was just going for something to be a little smart just to avoid adding a new setting or upsetting users who use Mode 2 in a field and wanted the lighting. 

I'm very cool with just locking Mode 2 to headlights if that's a better solution.

Reply:
Oh, and as far as manuring, like turns for the fieldworkAIDriver, that one does not use an is field check.